### PR TITLE
Expand Sql test coverage #2623

### PR DESCRIPTION
**Pull request by JiriOndrusek**
https://github.com/apache/camel-quarkus/issues/2623

This PR does not fix the issue completely. Another native issue was detected and reported as https://github.com/apache/camel-quarkus/issues/2693.

Test covering https://github.com/apache/camel-quarkus/issues/2693 is present, but is disabled. (`SqlTest.testAggregationRepository`)

But this could be merged. New issue and test enablement would be solved as https://github.com/apache/camel-quarkus/issues/2693.

### CODE REVIEW

File: integration-tests/sql/src/main/java/org/apache/camel/quarkus/component/sql/it/SqlRoutes.java [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+        File tmpFile = File.createTempFile("selectProjects-", ".sql");
+        tmpFile.deleteOnExit();
+        try (InputStream is = Thread.currentThread().getContextClassLoader().getResourceAsStream(file)) { //todo quick workaround
+
+            ByteArrayOutputStream out = new ByteArrayOutputStream();
+            int c;
+            while ((c = is.read()) >= 0) {
+                out.write(c);
+            }
+            new FileOutputStream(tmpFile).write(out.toByteArray());
```
Comment (ppalaga) [inline integration-tests/sql/src/main/java/org/apache/camel/quarkus/component/sql/it/SqlRoutes.java:116]: Not closing a FileOutputStream may lead to surprising reads and perhaps also file being locked on Windows. `java.nio.file.Files.write(Path, byte[], OpenOption...)` might be a good replacement for this line, or even better the whole try block could be replaced by  `java.nio.file.Files.copy(InputStream, Path, CopyOption...)`

File: integration-tests/sql/src/test/java/org/apache/camel/quarkus/component/sql/it/SqlTest.java [hunk 2]
```diff
@@ -66,4 +78,169 @@ public void testSqlStoredComponent() {
...
+        Map project = CollectionHelper.mapOf("ID", id, "PROJECT", routeId, "LICENSE", "222", "PROCESSED", false);
+        Map updatedProject = CollectionHelper.mapOf("ID", id, "PROJECT", routeId, "LICENSE", "XXX", "PROCESSED", false);
+
+        postMapWithParam("/sql/insert",
+                "table", "projects",
+                project)
+                        .statusCode(201);
+
+        //wait for the record to be caught
+        Thread.sleep(500);
```
Comment (ppalaga) [inline integration-tests/sql/src/test/java/org/apache/camel/quarkus/component/sql/it/SqlTest.java:109]: Could we please await? 500 ms might be too much on a laptop and too little on a CI.

File: integration-tests/sql/src/test/java/org/apache/camel/quarkus/component/sql/it/SqlTest.java [hunk 3]
```diff
@@ -66,4 +78,169 @@ public void testSqlStoredComponent() {
...
+                .statusCode(200)
+                .body("size()", is(1), "$", hasItem(project));
+
+        //update
+        postMapWithParam("/sql/update",
+                "table", "projects",
+                updatedProject)
+                        .statusCode(201);
+
+        Thread.sleep(500);
```
Comment (ppalaga) [inline integration-tests/sql/src/test/java/org/apache/camel/quarkus/component/sql/it/SqlTest.java:122]: Could we please await? 500 ms might be too much on a laptop and too little on a CI.

File: integration-tests/sql/src/test/java/org/apache/camel/quarkus/component/sql/it/SqlTest.java [hunk 4]
```diff
@@ -66,4 +78,169 @@ public void testSqlStoredComponent() {
...
+                .statusCode(200);
+
+        postMapWithParam("/sql/toDirect/aggregation", "body", "B", CollectionHelper.mapOf("messageId", "123"))
+                .statusCode(200);
+
+        postMapWithParam("/sql/toDirect/aggregation", "body", "C", CollectionHelper.mapOf("messageId", "123"))
+                .statusCode(200);
+
+        postMapWithParam("/sql/toDirect/aggregation", "body", "D", CollectionHelper.mapOf("messageId", "123"))
+                .statusCode(200);
```
Comment (ppalaga) [inline integration-tests/sql/src/test/java/org/apache/camel/quarkus/component/sql/it/SqlTest.java:209]: Not packing the RestAssured code into a method would make it easier to understand what this actually does. Maybe it is just me? What do others think?

**Review by ppalaga — CHANGES_REQUESTED** [review]
Thanks, @JiriOndrusek, nice coverage. Some questions and requests inline.

File: integration-tests/sql/src/main/java/org/apache/camel/quarkus/component/sql/it/SqlRoutes.java (hunk 1 above)
Comment (JiriOndrusek) [reply integration-tests/sql/src/main/java/org/apache/camel/quarkus/component/sql/it/SqlRoutes.java:116]: My fault, I'll fix it.

File: integration-tests/sql/src/test/java/org/apache/camel/quarkus/component/sql/it/SqlTest.java (hunk 2 above)
Comment (JiriOndrusek) [reply integration-tests/sql/src/test/java/org/apache/camel/quarkus/component/sql/it/SqlTest.java:109]: Yes, I'll do it, awat is a much better solution

File: integration-tests/sql/src/test/java/org/apache/camel/quarkus/component/sql/it/SqlTest.java (hunk 3 above)
Comment (JiriOndrusek) [reply integration-tests/sql/src/test/java/org/apache/camel/quarkus/component/sql/it/SqlTest.java:122]: Yes, I'll do it, awat is a much better solution

File: integration-tests/sql/src/test/java/org/apache/camel/quarkus/component/sql/it/SqlTest.java (hunk 4 above)
Comment (JiriOndrusek) [reply integration-tests/sql/src/test/java/org/apache/camel/quarkus/component/sql/it/SqlTest.java:209]: I tried to keep all parametetrs (like url, body, params) in the main call (not helper method) - from my POV it seems more understandable and shows that all the calls are the same.
But I don't have any problem with keeping restassured  code intact...

Comment (ppalaga) [reply integration-tests/sql/src/test/java/org/apache/camel/quarkus/component/sql/it/SqlTest.java:209]: Plz. do not change anything, maybe it is just me :)

File: integration-tests/sql/src/main/java/org/apache/camel/quarkus/component/sql/it/SqlRoutes.java (hunk 1 above)
Comment (JiriOndrusek) [reply integration-tests/sql/src/main/java/org/apache/camel/quarkus/component/sql/it/SqlRoutes.java]: fixed

File: integration-tests/sql/src/test/java/org/apache/camel/quarkus/component/sql/it/SqlTest.java (hunk 3 above)
Comment (JiriOndrusek) [reply integration-tests/sql/src/test/java/org/apache/camel/quarkus/component/sql/it/SqlTest.java]: fixed

File: integration-tests/sql/src/test/java/org/apache/camel/quarkus/component/sql/it/SqlTest.java (hunk 2 above)
Comment (JiriOndrusek) [reply integration-tests/sql/src/test/java/org/apache/camel/quarkus/component/sql/it/SqlTest.java]: fixed

**Comment by JiriOndrusek** [comment]
@ppalaga All items except the last one are fixed.

**Review by ppalaga — APPROVED** [review]
Thanks, @JiriOndrusek !

**Pull request merged**